### PR TITLE
iox-#136 Rename examples from '*_on_c' to '*_in_c'

### PR DESCRIPTION
**Pull request by mossmaurice**
Signed-off-by: Simon Hoinkis <simon.hoinkis@apex.ai>

## Pre-Review Checklist for the PR Author

1. [x] Branch follows the naming format (`iox-#123-this-is-a-branch`)
1. [x] Commits messages are according to this [guideline][commit-guidelines]
    - [x] Commit messages have the issue ID (`iox-#123 commit text`)
    - [x] Commit messages are signed (`git commit -s`)
    - [x] Commit author matches [Eclipse Contributor Agreement][eca] (and ECA is signed)
1. [x] Update the PR title
   - Follow the same conventions as for commit messages
   - Link to the relevant issue
1. [x] Relevant issues are linked
1. [x] Add sensible notes for the reviewer
1. [x] All checks have passed
1. [x] Assign PR to reviewer

[commit-guidelines]: https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
[eca]: http://www.eclipse.org/legal/ECA.php

## Notes for Reviewer
* Rename examples from `*_on_c` to `*_in_c`

## Checklist for the PR Reviewer

- [x] Commits are properly organized and messages are according to the guideline
- [x] Code according to our coding style and naming conventions
- [x] Unit tests have been written for new behavior
- [x] Public API changes are documented via doxygen
- [x] Copyright owner are updated in the changed files
- [x] PR title describes the changes

## Post-review Checklist for the PR Author

1. [x] All open points are addressed and tracked via issues

## Post-review Checklist for the Eclipse Committer

1. [x] All checkboxes in the PR checklist are checked or crossed out
1. [ ] Merge

## References

- Closes partly #136

### CODE REVIEW

**Review by elBoberido — DISMISSED** [review]

File: iceoryx_examples/icedelivery_in_c/topic_data.h [hunk 1]
```diff
@@ -12,8 +12,8 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-#ifndef IOX_EXAMPLES_ICEDELIVERY_ON_C_TOPIC_DATA_HPP
-#define IOX_EXAMPLES_ICEDELIVERY_ON_C_TOPIC_DATA_HPP
+#ifndef IOX_EXAMPLES_ICEDELIVERY_in_c_TOPIC_DATA_HPP
```
Comment (elfenpiff) [inline iceoryx_examples/icedelivery_in_c/topic_data.h:15]: How I love `#pragma once` and hate these macro guards and on that note: They have to be all CAPITAL_SNAKE_CASE and you forgot `_in_c_`.

**Review by elfenpiff — CHANGES_REQUESTED** [review]
INCLUDE_GUARDS_are_wrong

File: iceoryx_examples/icedelivery_in_c/topic_data.h (hunk 1 above)
Comment (mossmaurice) [reply iceoryx_examples/icedelivery_in_c/topic_data.h:15]: Thanks for this catch. That's why one should always double check a search and replace...

Comment (mossmaurice) [reply iceoryx_examples/icedelivery_in_c/topic_data.h]: ..and use match case. Fixed.

**Review by elBoberido — APPROVED** [review]

**Comment by elBoberido** [comment]
> INCLUDE_GUARDS_are_wrong

having this as automatic check could be a good task to start contributing

**Review by ithier — APPROVED** [review]

**Pull request merged**